js_src/header.ts

```diff
@@ -0,0 +1,73 @@
+export default class Header {
```

Comment:
seems like weve got a lot of code duplication here between the previous file. Recognising we are trying to move fast here but it might be a sign we need to extract some this functionality out and share it.

Reply:
It might be possible to create a parent class "AppliedToElement", from which the "find" method can be inherited.